primitives/state-machine/src/lib.rs

```diff
@@ -204,8 +204,10 @@ impl<'a, B, H, N, T, Exec> StateMachine<'a, B, H, N, T, Exec> where
 		exec: &'a Exec,
 		method: &'a str,
 		call_data: &'a [u8],
-		extensions: Extensions,
+		mut extensions: Extensions,
 	) -> Self {
+		extensions.register(CallInWasmExt::new(exec.clone()));
```

Comment:
So it's now available to every runtime call I guess, what are the consequences here? Besides using it being super slow, as I guess a new wasm env has to be initiated.

Reply:
Why super slow? The wasm blob needs to be initialized when we want to call some function in it, but there is nothing we can optimize (as we don't know the blob in front of).

We need the extension while building/executing a block. But there is also no direct method that you can call from the runtime, besides the one that only can call `Core_version`.

Reply:
Sure, I know there is no other way to call `Core_version` from a different runtime blob, but afair when talking with Sergei he was saying that initializing wasm runtime (alloc memory, load module, etc) takes quite some time, hence I've mentioned that it's most likely going to be way slower than any other operation or host function we have, so we need to be careful to avoid using it too often.
I was wondering if there are any other (more severe) consequences of exposing this kind of function to the runtime - i.e. if it can be somehow abused.

> . But there is also no direct method that you can call from the runtime, besides the one that only can call Core_version.

That's good to know, so it means that it's not easy to shoot yourself in a foot if you are a runtime developer, right?

Reply:
Yeah it is probably no easy operation, but better to include less transactions than to stop the network again ;) If something does not work or for whatever reason we need to put a blob in without any checks, I added `set_code_without_checks`. This will just do it the old way, so we are sure that we can put a blob in :) 

Yeah, the runtime developer can not access the `CallInRuntimeExt` in any other way. It is not exposed to the runtime. The developer can just call `misc::runtime_version` that uses this stuff internally.

Reply:
Okay, that's fair, but I would really add extra docs to make sure it's not being called as it's a possible DoS vector. Imagine if we do that check whenever someone proposes a runtime, then people can craft arbitrary runtimes and do arbitrary operations within `Core_version` and it wouldn't be priced correctly. So we should only make sure to run in during `set_code`, not during any preliminary checks (that accept unreviewed user's input).